cosmoz-grouped-list.js

```diff
@@ -231,12 +235,32 @@
 				});
 
 				if (group.items.length) {
-					return flatData.concat(group, group.items);
+					return acc.concat(group, group.items);
 				} else if (this.displayEmptyGroups) {
-					return flatData.concat(group);
+					return acc.concat(group);
 				}
-				return flatData;
+				return acc;
 			}, []);
+
+			this.selectedItems = this.selectedItems.filter(i => flatData.includes(i));
+
+			return flatData;
+		}
+
+		_assertDataIsNotMixed(data) {
+			if (!data || data.length === 0) {
+				return;
+			}
+
+			var shouldBeAllGroups = data[0].items instanceof Array,
```

Comment:
Tough one!
I want to make it more straight forward, left a suggestion.
We also try to avoid negating stuff, but mixed vs coherent is not really negated.
But generally easier to try to explain the expected positive scenario and then `if (!ok) { throw Error(); }`.

Anyway at least never use `var` and consider the suggestion!
Also, we normally use `Array.isArray()` rather than `instanceof`, so I think we should stick to that until we decide otherwise. Same above instead of `!data`.

```suggestion
			const shouldBeGrouped = Array.isArray(data[0].items),
				isCoherent = data.every(group => Array.isArray(group.items) === shouldBeGrouped);
```

Reply:
Thank you for the suggestion! I should have reviewed the code a bit more before pushing.